lib/CMakeLists.txt

```diff
@@ -13,487 +12,91 @@
-endif()
-if(IS_MACOSX)
-    list(APPEND compile_definitions_list OSMac_)
-endif()
-
-target_compile_definitions(${LIBRARY_NAME} PRIVATE
-    ${compile_definitions_list}
+# -----------------------------------------------------------------------------
+# compile definitions
+# -----------------------------------------------------------------------------
+target_compile_definitions(${PROJECT_NAME} 
+    PRIVATE
+        MAYAUSD_MACROS_EXPORT
+        MAYAUSD_CORE_EXPORT
+        MFB_PACKAGE_NAME="${PROJECT_NAME}"
+        MFB_ALT_PACKAGE_NAME="${PROJECT_NAME}"
+        MFB_PACKAGE_MODULE=mayaUsd
+        $<$<BOOL:${UFE_FOUND}>:WANT_UFE_BUILD>
+        $<$<BOOL:${IS_MACOSX}>:OSMac_>
 )
 
 # Some of the UFE classes are exporting STL classes which causes this warning.
 if(UFE_FOUND AND MSVC)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4251")
     string(REPLACE " " ";" REPLACED_FLAGS ${CMAKE_CXX_FLAGS})
-    target_compile_options(${LIBRARY_NAME} INTERFACE ${REPLACED_FLAGS})
-endif()
-
-set(include_directories_list
-    ${MAYA_INCLUDE_DIRS}
-    ${PXR_INCLUDE_DIRS}
-    ${CMAKE_BINARY_DIR}/include
-)
-
-if(UFE_FOUND)
-    list(APPEND include_directories_list ${UFE_INCLUDE_DIR})
-endif()
-
-target_include_directories( ${LIBRARY_NAME} PUBLIC
-    ${include_directories_list}
-)
-
-target_link_libraries(${LIBRARY_NAME}
-    ${Boost_FILESYSTEM_LIBRARY}
-    ${Boost_SYSTEM_LIBRARY}
-    ${MAYA_Foundation_LIBRARY}
-    ${MAYA_OpenMayaAnim_LIBRARY}
-    ${MAYA_OpenMaya_LIBRARY}
-    ${MAYA_OpenMayaRender_LIBRARY}
-    ${MAYA_OpenMayaUI_LIBRARY}
-    ar
-    gf
-    hd
-    hdx
-    js
-    kind
-    plug
-    sdf
-    tf
-    trace
-    usd
-    usdGeom
-    usdImaging
-    usdImagingGL
-    usdLux
-    usdRi
-    usdShade
-    usdSkel
-    usdUtils
-    vt
+    target_compile_options(${PROJECT_NAME} INTERFACE ${REPLACED_FLAGS})
+endif()
+
+# -----------------------------------------------------------------------------
+# include directories
+# -----------------------------------------------------------------------------
+target_include_directories(${PROJECT_NAME} 
+    PUBLIC
+        ${CMAKE_BINARY_DIR}/include
+        ${MAYA_INCLUDE_DIRS}
+        ${PXR_INCLUDE_DIRS}
+        $<$<BOOL:${UFE_FOUND}>:${UFE_INCLUDE_DIR}>
+    PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
+# -----------------------------------------------------------------------------
+# link libraries
+# -----------------------------------------------------------------------------
+target_link_libraries(${PROJECT_NAME}
+    PUBLIC
+        ar
+        gf
+        hd
+        hdx
+        js
+        kind
+        plug
+        sdf
+        tf
+        trace
+        usd
+        usdGeom
+        usdImaging
+        usdImagingGL
+        usdLux
+        usdRi
+        usdShade
+        usdSkel
+        usdUtils
+        vt
+        $<$<BOOL:${UFE_FOUND}>:${UFE_LIBRARY}>
+        ${MAYA_LIBRARIES}
+    PRIVATE
+        ${Boost_FILESYSTEM_LIBRARY}
+        ${Boost_SYSTEM_LIBRARY}
```

Comment:
Can we start using the named boost targets now, like `Boost::filesystem` and `Boost::system`?

Reply:
Good idea. I like your suggestion. Will fix it in the next commit.

Reply:
see e25fa484